src/azure-cli/azure/cli/command_modules/network/operations/load_balancer.py

```diff
@@ -126,6 +128,74 @@ def post_instance_update(self, instance):
             instance.properties.gateway_load_balancer = None
 
 
+class LBInboundNatPoolCreate(_LBInboundNatPoolCreate):
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.frontend_ip_name = AAZStrArg(
+            options=["--frontend-ip-name"],
+            arg_group="Properties",
+            help="The name the frontend IP configuration. If only one exists, omit to use as default.",
+        )
+
+        args_schema.frontend_ip._registered = False
+        return args_schema
+
+    def pre_instance_create(self):
+        args = self.ctx.args
+        instance = self.ctx.vars.instance
+        if has_value(args.frontend_ip_name):
+            frontend_ip_name = args.frontend_ip_name.to_serialized_data()
+            for frontend_ip in instance.properties.frontend_ip_configurations:
+                if frontend_ip.name == frontend_ip_name:
+                    args.frontend_ip = frontend_ip.id
+                    break
+            if not has_value(args.frontend_ip):
+                raise InvalidArgumentValueError(
+                    "FrontendIpConfiguration '{}' does not exist".format(frontend_ip_name))
+        elif has_value(instance.properties.frontend_ip_configurations) and len(instance.properties.frontend_ip_configurations) == 1:
+            args.frontend_ip = instance.properties.frontend_ip_configurations[0].id
+
+
+class LBInboundNatPoolUpdate(_LBInboundNatPoolUpdate):
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.frontend_ip_name = AAZStrArg(
+            options=["--frontend-ip-name"],
+            arg_group="Properties",
+            help="The name the frontend IP configuration.",
+            nullable=True
+        )
+
+        args_schema.frontend_ip._registered = False
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        args = self.ctx.args
+        instance = self.ctx.vars.instance
+        if has_value(args.frontend_ip_name):
+            frontend_ip_name = args.frontend_ip_name.to_serialized_data()
+            if frontend_ip_name is None:
+                args.frontend_ip = None
+            else:
+                for frontend_ip in instance.properties.frontend_ip_configurations:
+                    if frontend_ip.name == frontend_ip_name:
+                        args.frontend_ip = frontend_ip.id
+                        break
+                if not has_value(args.frontend_ip):
+                    raise InvalidArgumentValueError(
+                        "FrontendIpConfiguration '{}' does not exist".format(frontend_ip_name))
```

Comment:
If I understand correctly, when `frontend_ip_name` is "", there is an exception raising instead of detaching the frontend IP silently?

Reply:
I tried the "" value, the service response error as frontend_ip is required property and it's not able to be detached. It's why correct the previous logic.